web/source/osk/embedded/subkeyDelegator.ts

```diff
@@ -1,16 +1,53 @@
 namespace com.keyman.osk.embedded {
   export class SubkeyDelegator {
     private resolver: (keyEvent: text.KeyEvent) => void;
+    private readonly vkbd: VisualKeyboard;
 
     public readonly baseKey: KeyElement;
+    public readonly promise: Promise<text.KeyEvent>;
+
+    constructor(vkbd: VisualKeyboard, e: KeyElement) {
+      this.vkbd = vkbd;
+
+      let _this = this;
+      this.promise = new Promise<text.KeyEvent>(function(resolve) {
+        _this.resolver = resolve;
+      });
 
-    constructor(e: KeyElement, resolve: (keyEvent: text.KeyEvent) => void) {
-      this.resolver = resolve;
       this.baseKey = e;
     }
 
-    public resolve(keyEvent: text.KeyEvent) {
+    public resolve(keyCoreID: string) {
       if(this.resolver) {
+        let keyEvent: text.KeyEvent = null;
+
+        if(keyCoreID != null) {
+          // This is set with the base key of our current subkey elsewhere within the engine.
+          var baseKey: OSKKeySpec = this.baseKey.key.spec;
```

Comment:
```suggestion
          let baseKey: OSKKeySpec = this.baseKey.key.spec;
```

_let_'s use consistent variable declarations in new code 🤣